src/transformers/trainer.py

```diff
@@ -2419,16 +2420,22 @@ def prediction_step(
                     logits = smp_nested_concat(logits_mb)
             else:
                 if has_labels:
-                    loss, outputs = self.compute_loss(model, inputs, return_outputs=True)
-                    loss = loss.mean().detach()
+                    if self.use_amp:
+                        with autocast(dtype=self.amp_dtype):
+                            loss, outputs = self.compute_loss(model, inputs, return_outputs=True)
+                            loss = loss.mean().detach()
```

Comment:
Are you trying to add autocast for inference? Could you please explain why?

Reply:
(additionally you probably will want to rebase against master at some point as this branch has a conflicting file)

Reply:
>rebase against master

Done, thanks

>autocast for inference

Oh, that's a good question.  I don't think that should be there.

Reply:
What's the justification for autocasting [here](https://github.com/huggingface/transformers/pull/13207/files/0770fca411ca9f28eefb8b06b8886b215e27ca06..c108936ddd767e365931a371ae1d642851581b16#diff-ed55888e6665791fe92cc8fc0c499da54f4ace6738551cd9a2591881cda076deL2431)?

Reply:
I checked in with @sgugger and it looks like I was wrong. 

I was under impression that one won't want to use AMP for inference and go for .`half()` instead if they wanted fp16. but because of `LayerNorm` which ideally should be in fp32, AMP is used instead.

it's questionable how efficient that is, since it has to recast to fp16 (as compared to `.half()`), but with autocast cache it's still faster than doing inference in pure-fp32. I think it uses more memory as well, as it has both copies.

So you have uncovered a missing feature there, James.

Would it be possible to add the 2423-2424 lines in a separate PR so not to mix it with bf16 extension?

Thank you!